File: crowdin/src/main/java/com/crowdin/platform/utils/TextUtils.java
package com.crowdin.platform.utils;

import android.content.res.Resources;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.util.SparseArray;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.crowdin.platform.transformers.MenuItemStrings;

public class TextUtils {

    private TextUtils() {
    }

    @Nullable
    public static String getTextForAttribute(AttributeSet attrs, int index, View view, Resources resources) {
        String text = null;
        String value = attrs.getAttributeValue(index);
        if (value != null && value.startsWith("@")) {
            text = resources.getString(attrs.getAttributeResourceValue(index, 0));
        }

        return text;
    }

    public static void updateMenuItemsText(Menu menu, Resources resources, int resId) {
        SparseArray<MenuItemStrings> itemStrings = XmlParserUtils.getMenuItemsStrings(resources, resId);

        for (int i = 0; i < itemStrings.size(); i++) {
            int itemKey = itemStrings.keyAt(i);
            MenuItemStrings itemValue = itemStrings.valueAt(i);

            if (itemValue.getTitle() != 0) {
                MenuItem menuItem = menu.findItem(itemKey);
                if (menuItem != null) {
                    menuItem.setTitle(resources.getString(itemValue.getTitle()));
                }
            }
            if (itemValue.getTitleCondensed() != 0) {
                MenuItem menuItem = menu.findItem(itemKey);
                if (menuItem != null) {
                    menuItem.setTitleCondensed(resources.getString(itemValue.getTitleCondensed()));
                }
            }
        }
    }
}

File: crowdin/src/main/java/com/crowdin/platform/utils/TextUtils.kt
package com.crowdin.platform.utils

import android.content.res.Resources
import android.util.AttributeSet
import android.view.Menu

internal object TextUtils {

    fun getTextForAttribute(attrs: AttributeSet, index: Int, resources: Resources): CharSequence? {
        var text: CharSequence? = null
        val value = attrs.getAttributeValue(index)
        if (value != null && value.startsWith("@")) {
            text = resources.getText(attrs.getAttributeResourceValue(index, 0))
        }

        return text
    }

    fun updateMenuItemsText(menu: Menu, resources: Resources, resId: Int) {
        val itemStrings = XmlParserUtils.getMenuItemsStrings(resources, resId)

        for (i in 0 until itemStrings.size()) {
            val itemKey = itemStrings.keyAt(i)
            val itemValue = itemStrings.valueAt(i)

            if (itemValue.title != 0) {
                menu.findItem(itemKey)?.title = resources.getString(itemValue.title)
            }
            if (itemValue.titleCondensed != 0) {
                menu.findItem(itemKey)?.titleCondensed = resources.getString(itemValue.titleCondensed)
            }
        }
    }
}
